### PR TITLE
More cli tests

### PR DESCRIPTION
**Pull request by sibiraj-s**
- add tests for some common cases
- prefer native mkdir recursion to manual recursion

### CODE REVIEW

File: tests/cli.spec.js [hunk 1]
```diff
@@ -1,18 +1,33 @@
...
   return data;
 };
 
+const existsFixutre = (filePath) => {
+  return fs.existsSync(path.resolve(fixturesDir, filePath));
+};
+
+const removeFixture = async (p) => {
+  const pathToDelete = path.resolve(fixturesDir, p);
+  // TODO: Temp fix for CI running on node 12, since
```
Comment (DanielRuf) [inline tests/cli.spec.js:22]:
```suggestion
  // TODO: Temp fix for CI running on node 12
```

File: tests/cli.spec.js [hunk 2]
```diff
@@ -1,18 +1,33 @@
...
+  // since fs.rm is available only since node 14
```
Comment (DanielRuf) [inline tests/cli.spec.js:23]:
```suggestion
  // because fs.rm is available only in node 14 and newer
```

File: tests/fixtures/url.html [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+<head>
+  <meta charset="UTF-8">
+  <meta http-equiv="X-UA-Compatible" content="IE=edge">
+  <meta name="viewport" content="width=device-width, initial-scale=1.0">
+  <title>Document</title>
+</head>
+
+<body>
+  <link rel="stylesheet" href="http://website.com/style.css">
+  <form action="http://website.com/folder/folder2/index.html">
```
Comment (DanielRuf) [inline tests/fixtures/url.html:13]:
```suggestion
  <form action="http://example.com/folder/folder2/index.html">
```

File: tests/fixtures/url.html [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+<head>
+  <meta charset="UTF-8">
+  <meta http-equiv="X-UA-Compatible" content="IE=edge">
+  <meta name="viewport" content="width=device-width, initial-scale=1.0">
+  <title>Document</title>
+</head>
+
+<body>
+  <link rel="stylesheet" href="http://website.com/style.css">
```
Comment (DanielRuf) [inline tests/fixtures/url.html:12]:
```suggestion
  <link rel="stylesheet" href="http://example.com/style.css">
```

File: tests/fixtures/url.html [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+  <form action="http://website.com/folder/folder2/index.html">
+    <a href="http://website.com/folder/file.html">link</a>
```
Comment (DanielRuf) [inline tests/fixtures/url.html:14]:
```suggestion
    <a href="http://example.com/folder/file.html">link</a>
```

**Review by DanielRuf — CHANGES_REQUESTED** [review]

File: tests/cli.spec.js (hunk 1 above)
Comment (sibiraj-s) [reply tests/cli.spec.js]: done.

File: tests/cli.spec.js (hunk 2 above)
Comment (sibiraj-s) [reply tests/cli.spec.js]: done

File: tests/fixtures/url.html (hunk 3 above)
Comment (sibiraj-s) [reply tests/fixtures/url.html]: done.

File: tests/fixtures/url.html (hunk 4 above)
Comment (sibiraj-s) [reply tests/fixtures/url.html]: done

File: tests/fixtures/url.html (hunk 5 above)
Comment (sibiraj-s) [reply tests/fixtures/url.html]: done

**Comment by sibiraj-s** [comment]
Lets merge this?

**Review by DanielRuf — APPROVED** [review]

**Pull request merged**